picard/ui/infodialog.py

```diff
@@ -88,8 +92,56 @@ def _display_artwork_tab(self):
             if image.width and image.height:
                 infos.append(u"%d x %d" % (image.width, image.height))
             infos.append(image.mimetype)
-            item.setText(u"\n".join(infos))
-            self.ui.artwork_list.addItem(item)
+
+            img_wgt = self.artwork_table.get_coverart_widget(pixmap, "\n".join(infos))
+            self.artwork_table.setCellWidget(row, col, img_wgt)
+            self.artwork_table.setItem(row, col, item)
+            row += 1
+
+    def _display_artwork_type(self):
+        types = []
+        types = [image.types_as_string() for image in self.obj.metadata.images]
+        if self.display_existing_artwork:
+            existing_types = [image.types_as_string() for image in self.obj.orig_metadata.images]
+            #Merge both lists
+            temp = []
+            i=0
+            j=0
+            while i!=len(types) and j!=len(existing_types):
+                if types[i] > existing_types[j]:
+                    temp.append(existing_types[j])
+                    j+=1
+                elif types[i] < existing_types[j]:
+                    temp.append(types[i])
+                    i+=1
+                else:
+                    temp.append(types[i])
+                    i+=1
+                    j+=1
+            if i==len(types):
+                temp.extend(existing_types[j:])
+            else:
+                temp.extend(types[i:])
+            types = temp
+        for row, type in enumerate(types):
+            self.artwork_table.insertRow(row)
+            item = QtGui.QTableWidgetItem()
+            item.setText(type)
+            item.setTextAlignment(QtCore.Qt.AlignCenter)
+            self.artwork_table.setItem(row, 0, item)
+
+    def arrange_images(self):
+        self.obj.metadata.images.sort(key=lambda img: img.types_as_string())
+        if self.display_existing_artwork:
+            self.obj.orig_metadata.images.sort(key=lambda img: img.types_as_string())
```

Comment:
Since the same lambda function is used twice, what about defining it once instead ?
Python permits locally defined functions.